### ISSUE TITLE
Allow to configure the optimisation parameters from `.yaml` file

### ISSUE THREAD

**Issue by ioangatop** [Lightning-AI/lightning#18107]
### Description & Motivation

Usually in self-supervised setting, like DINO, we would like to split the optimisation parameters into regulated and not regulated by weight decay. In these setting, we have to use the `configure_optimizers` method in `LightningModule`, and thus bypass the `optimizer` and `lr_scheduler` fields in the `.yaml` configuration file. The issue is that we loose the `jsonargparser` great feature to set objects from the `yaml` file and have to introduce a lot of logic in the `configure_optimizers` if we want to have configurable optimizers, but also, introduce a lot of parameters in the `LightningModule`'s `__init__` method.

### Pitch

Currently, in the CLI the optimizer get the [self.model.parameters()](https://github.com/Lightning-AI/lightning/blob/master/src/lightning/pytorch/cli.py#L648) by default. I think it would be great if we introduce an method in the `LightningModule` like `optimizer_parameters` which can be overwritten and customised - something that is usually standard for specific models. Then, we could still use the `yaml` file for optimizers and schedulers freely and act only in the parameters that we select.

The `optimizer_parameters` method can either (i) return `None` by default and the CLI will get the self.model.parameters() or (ii) return the `self.model.parameters()` and can be overwritten with custom logic 

```
class MyModelModule(pl.LightningModule):
    def optimizer_parameters(self) -> List[Dict["params", "weight_decay"]]:
        # see https://github.com/Lightning-Universe/lightning-bolts/blob/4f910f6c9893eaf206eed6b7df545c8a02043a55/src/pl_bolts/models/self_supervised/simsiam/simsiam_module.py#L154
        params, params_no_weight_decay = exclude_from_weight_decay(self.models.named_parameters())
        return [{"params": params}, {"params": params_no_weight_decay, "weight_decay": 0.0}]
```

cc @borda @carmocca @mauvilsa @justusschock @awaelchli

**Comment by mauvilsa** [comment]
This is already possible, see https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#multiple-optimizers-and-schedulers

**Comment by ioangatop** [comment]
> This is already possible, see https://lightning.ai/docs/pytorch/stable/cli/lightning_cli_advanced_3.html#multiple-optimizers-and-schedulers

@mauvilsa thank you for your response, I see so I should put the optimizer in the `LightningModule` and then link it to the `optimizer` in `yaml`?

**Comment by mauvilsa** [comment]
> so I should put the optimizer in the `LightningModule` and then link it to the `optimizer` in `yaml`?

I am not sure what you mean by link here. There wouldn't be any need for argument links.

> I would prefer to keep the optimizer completely outside of the `LightningModule` but it is a solution that I can work with

If you mean that in the `yaml`, the optimizer not be nested inside the model section, I guess you can also do that if you like. Just add an argument with type `OptimizerCallable` and then link that to the `LightningModule`'s `optimizer` parameter. Disclaimer, I haven't tried this.

**Comment by ioangatop** [comment]
@mauvilsa thank you for your response; just FYI I'm not able to link it because it seems the optimizer and scheduler are set after the link process in the CLI takes place
